src/main/java/picard/sam/util/SamComparison.java

```diff
@@ -300,9 +401,10 @@ private AlignmentComparison compareAlignmentRecords(final SAMRecord s1, final SA
     }
 
     private boolean alignmentsMatch(final SAMRecord s1, final SAMRecord s2) {
-        return (s1.getReferenceName().equals(s2.getReferenceName()) &&
+        return ((s1.getReferenceName().equals(s2.getReferenceName()) &&
                 s1.getAlignmentStart() == s2.getAlignmentStart() &&
-                s1.getReadNegativeStrandFlag() == s1.getReadNegativeStrandFlag());
+                s1.getReadNegativeStrandFlag() == s1.getReadNegativeStrandFlag()) ||
+                (mq0Match && s1.getMappingQuality() <= 3 && s2.getMappingQuality() <= 3));
```

Comment:
I think MQ=255 reads should be treated separately, since both the SAM spec and the way aligners tend to actually use MQ treat low MQ and MQ 255 as very different situations.  So have added a separate argument to control behavior for MQ 255 reads.